### PR TITLE
MockTracer onSpanFinished

### PR DESCRIPTION
**Pull request by pavolloffay**
@bensigelman @yurishkuro could you please review? 

https://github.com/opentracing/opentracing-java/issues/74

### CODE REVIEW

File: opentracing-mock/src/main/java/io/opentracing/mock/MockTracer.java [hunk 1]
```diff
@@ -65,6 +65,12 @@ public synchronized void reset() {
     }
 
     /**
+     * Noop method called on {@link Span#finish()}.
+     */
+    public void onSpanFinished(MockSpan mockSpan) {
```
Comment (yurishkuro) [inline opentracing-mock/src/main/java/io/opentracing/mock/MockTracer.java]: I think you want `protected`

File: opentracing-mock/src/main/java/io/opentracing/mock/MockSpan.java [hunk 2]
```diff
@@ -96,6 +96,7 @@ public void finish() {
     public synchronized void finish(long finishMicros) {
         this.finishMicros = finishMicros;
         this.mockTracer.appendFinishedSpan(this);
+        this.mockTracer.onSpanFinished(this);
```
Comment (yurishkuro) [inline opentracing-mock/src/main/java/io/opentracing/mock/MockSpan.java]: I think the tracer itself should call this from `appendFinishedSpan`

**Comment by pavolloffay** [comment]
PR updated.

**Comment by pavolloffay** [comment]
When do you plan to do next release? I will need it for `contrib/java-jaxrs`

**Comment by bensigelman** [comment]
@pavolloffay I'll try(*) to cut a release right now.

(*): we are using this clever TravisCI release process... the last time we tried to use it "something broke", though, so let's cross our collective fingers.

**Comment by pavolloffay** [comment]
@bensigelman @yurishkuro thanks!

**Comment by pavolloffay** [comment]
@bensigelman We missed  a quite important thing https://github.com/opentracing/opentracing-java/blob/master/opentracing-mock/src/main/java/io/opentracing/mock/MockTracer.java#L32 

I will create a PR to fix it.

**Pull request merged**